From 85cac39633b36d7274a1e33400b5e5d878c389ae Mon Sep 17 00:00:00 2001
From: Nikolay Orlyuk <virkony@gmail.com>
Date: Thu, 5 Jul 2012 23:01:16 +0300
Subject: [PATCH 1/3] Fix for bug 203 - DESTDIR brokes installation

See http://bugs.mercury.csse.unimelb.edu.au/view.php?id=203
Stripped DESTDIR when building boehm_gc/libatomic_ops
Added DESTDIR to default values in scripts/mmc
---
 boehm_gc/build_atomic_ops.sh        | 2 +-
 boehm_gc/build_atomic_ops.sh.cygwin | 2 +-
 scripts/mmc.in                      | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/boehm_gc/build_atomic_ops.sh b/boehm_gc/build_atomic_ops.sh
index faf19d0..638190b 100755
--- a/boehm_gc/build_atomic_ops.sh
+++ b/boehm_gc/build_atomic_ops.sh
@@ -3,4 +3,4 @@ P=`pwd`/libatomic_ops-install
 cd libatomic_ops
 # Mercury-specific: allow additional arguments.
 ./configure --prefix=$P "$@"
-$MAKE CC="$CC" install
+$MAKE CC="$CC" DESTDIR= install
diff --git a/boehm_gc/build_atomic_ops.sh.cygwin b/boehm_gc/build_atomic_ops.sh.cygwin
index 158a3f3..d9b963f 100755
--- a/boehm_gc/build_atomic_ops.sh.cygwin
+++ b/boehm_gc/build_atomic_ops.sh.cygwin
@@ -8,7 +8,7 @@ Q=`mktemp -d`
 ln -s "$P" $Q/dir
 cd $Q/dir/libatomic_ops
 ./configure --prefix=$Q/dir/libatomic_ops-install
-$MAKE CC="$CC" install
+$MAKE CC="$CC" DESTDIR= install
 cd /
 rm $Q/dir
 rmdir $Q
diff --git a/scripts/mmc.in b/scripts/mmc.in
index 510e018..d9d0e8b 100644
--- a/scripts/mmc.in
+++ b/scripts/mmc.in
@@ -14,8 +14,8 @@
 # MERCURY_COMPILER, MERCURY_C_COMPILER, MERCURY_DEFAULT_GRADE,
 # MERCURY_DEFAULT_OPT_LEVEL.
 
-MERCURY_COMPILER=${MERCURY_COMPILER-'@PREFIX@/bin/mercury_compile'}
-MERCURY_CONFIG_DIR=${MERCURY_CONFIG_DIR-${MERCURY_STDLIB_DIR-'@CONFIG_LIBDIR@'}}
+MERCURY_COMPILER=${MERCURY_COMPILER-${DESTDIR}'@PREFIX@/bin/mercury_compile'}
+MERCURY_CONFIG_DIR=${MERCURY_CONFIG_DIR-${DESTDIR}${MERCURY_STDLIB_DIR-'@CONFIG_LIBDIR@'}}
 export MERCURY_COMPILER MERCURY_CONFIG_DIR
 
 # Set the MACOSX_DEPLOYMENT_TARGET environment variable if needed.
-- 
1.8.0.3

